### PR TITLE
Issue 3332 - overlapping aliases should be prevented

### PR DESCRIPTION
**Pull request by bbakerman**
This shows the problem as reported in #3332 as happening

However I have another test that works and logically I cant see the difference.

Anyway here is a reproduction we can work on

### CODE REVIEW

File: src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy [hunk 1]
```diff
@@ -349,6 +350,58 @@ class OverlappingFieldsCanBeMergedTest extends Specification {
...
+        }
+        ''')
+        when:
+        traverse(query, schema)
+
+        then:
+        errorCollector.getErrors().size() == 1
+        errorCollector.getErrors()[0].message == "Validation error (FieldsConflict@[cat]) : 'name' : 'foo1' and 'foo2' are different fields"
+        errorCollector.getErrors()[0].locations == [new SourceLocation(3, 13), new SourceLocation(4, 13)]
+    }
```
Comment (bbakerman) [inline src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy:403]: This fails!

File: src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy [hunk 2]
```diff
@@ -349,6 +350,58 @@ class OverlappingFieldsCanBeMergedTest extends Specification {
...
+        }
+        type Query { dog: Dog }
+        ''')
+        when:
+        traverse(query, schema)
+
+        then:
+        errorCollector.getErrors().size() == 1
+        errorCollector.getErrors()[0].message == "Validation error (FieldsConflict@[dog]) : 'name' : 'nickname' and 'name' are different fields"
+        errorCollector.getErrors()[0].locations == [new SourceLocation(3, 13), new SourceLocation(4, 13)]
```
Comment (bbakerman) [inline src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy:374]: This does NOT fail ;  wat ??

Comment (bbakerman) [reply src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy:374]: `dog` parent field was nullable

File: src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy (hunk 1 above)
Comment (bbakerman) [reply src/test/groovy/graphql/validation/rules/OverlappingFieldsCanBeMergedTest.groovy:403]: and is now fixed

**Review by dondonz — APPROVED** [review]

**Pull request merged**